tests/test_ledger.py

```diff
@@ -1,445 +1,159 @@
+            ledger.add_ledger_entry(target)
+
+    def test_add_ledger_with_non_existing_account(self, ledger):
+        # Adding a ledger entry with non existing account should raise an error
+        target = self.LEDGER_ENTRIES.query("id == 1").copy()
+        target["account"] = 33333
+        with pytest.raises(ValueError, match="No id found for account"):
+            ledger.add_ledger_entry(target)
+
+    def test_add_ledger_with_non_existing_currency(self, ledger):
+        # Adding a ledger entry with non existing currency code should raise an error
+        target = self.LEDGER_ENTRIES.query("id == 1").copy()
+        target["currency"] = "Non_Existent_Currency"
+        with pytest.raises(ValueError, match="No id found for currency"):
+            ledger.add_ledger_entry(target)
+
+    @pytest.mark.parametrize("id", [15, 16])
+    def test_adding_transaction_with_two_non_base_currencies_fails(self, ledger, id):
+        # flake8: noqa: E501
+        LEDGER_CSV = """
+            id,     date, account, counter_account, currency,     amount, base_currency_amount, vat_code, text,                             document
+            15, 2024-06-26,     ,            9991,      USD,  100000.00,             90000.00,        ,   Convert 100k USD to EUR @ 0.9375,
+            15, 2024-06-26, 9990,                ,      EUR,   93750.00,             90000.00,        ,   Convert 100k USD to EUR @ 0.9375,
+            16, 2024-06-26,     ,            9991,      USD,  200000.00,            180000.00,        ,   Convert 200k USD to EUR and CHF,
+            16, 2024-06-26, 9990,                ,      EUR,   93750.00,             90000.00,        ,   Convert 200k USD to EUR and CHF,
+            16, 2024-06-26, 9992,                ,      CHF,   90000.00,             90000.00,        ,   Convert 200k USD to EUR and CHF,
+        """
+        # flake8: enable
+        target = pd.read_csv(StringIO(LEDGER_CSV), skipinitialspace=True)
+        target = self.LEDGER_ENTRIES[self.LEDGER_ENTRIES["id"] == id]
+        expected = (
+            "CashCtrl allows only the base currency plus a single foreign currency"
+        )
+        with pytest.raises(ValueError, match=expected):
+            ledger.add_ledger_entry(target)
+
+    def test_modify_non_existed_raises_error(self, ledger):
+        super().test_modify_non_existed_raises_error(
+            ledger, error_class=RequestException, error_message="entry does not exist"
+        )
+
+    def test_update_ledger_with_illegal_attributes(self, ledger):
+        # flake8: noqa: E501
+        LEDGER_CSV = """
+            id,     date, account, counter_account, currency,     amount, base_currency_amount, vat_code, text,                             document
+             1,  2024-05-24, 9992,            9995,      CHF,     100.00,                     ,  OutRed,   pytest single transaction 1,      /file1.txt
+        """
+        # flake8: enable
```

Comment:
Let's use a more compact version to reduce dependencies (to vat definitions) and comply with linting rules:
```
LEDGER_CSV = """
    date,       account, counter_account, currency, amount, text
    2024-05-24,    9992,            9995,      CHF, 100.00, Test
"""
```
instead of:
```
# flake8: noqa: E501
LEDGER_CSV = """
    id,     date, account, counter_account, currency,     amount, base_currency_amount, vat_code, text,                             document
     1,  2024-05-24, 9992,            9995,      CHF,     100.00,                     ,  OutRed,   pytest single transaction 1,      /file1.txt
"""
# flake8: enable
```